astroquery/mpc/core.py

```diff
@@ -274,34 +310,595 @@ def query_objects_async(self, target_type, get_query_payload=False, *args, **kwa
+            if hasattr(location, '__iter__'):
+                if len(location) != 3:
+                    raise ValueError(
+                        "location arrays require three values:"
+                        " longitude, latitude, and altitude")
+            else:
+                raise TypeError(
+                    "location must be a string, integer, array-like,"
+                    " or astropy EarthLocation")
+
+        if start is not None:
+            _start = Time(start)
+        else:
+            _start = None
+
+        # step must be one of these units, and must be an integer (we
+        # will convert to an integer later).  MPES fails for large
+        # integers, so we cannot just convert everything to seconds.
+        _step = u.Quantity(step)
+        if _step.unit not in [u.d, u.h, u.min, u.s]:
+            raise ValueError(
+                'step must have units of days, hours, minutes, or seconds.')
+
+        if number is not None:
+            if number > 1441:
+                raise ValueError('number must be <=1441')
+
+        if eph_type not in self._ephemeris_types.keys():
+            raise ValueError("eph_type must be one of {}".format(
+                self._ephemeris_types.keys()))
+
+        if proper_motion not in self._proper_motions.keys():
+            raise ValueError("proper_motion must be one of {}".format(
+                self._proper_motions.keys()))
+
+        if not u.Unit(proper_motion_unit).is_equivalent('rad/s'):
+            raise ValueError("proper_motion_unit must be an angular rate.")
+
+        # setup payload
+        request_args = self._args_to_ephemeris_payload(
+            target=target, ut_offset=ut_offset, suppress_daytime=suppress_daytime,
+            suppress_set=suppress_set, perturbed=perturbed, location=location,
+            start=_start, step=_step, number=number, eph_type=eph_type,
+            proper_motion=proper_motion)
+
+        # store for retrieval in _parse_result
+        self._ra_format = ra_format
+        self._dec_format = dec_format
+        self._proper_motion_unit = u.Unit(proper_motion_unit)
+        self._unc_links = unc_links
+
+        if get_query_payload:
+            return request_args
+
+        self.query_type = 'ephemeris'
+        response = self._request('POST', self.MPES_URL, data=request_args)
+
+        return response
+
+    @class_or_instance
+    def get_observatory_codes_async(self, get_raw_response=False, cache=True):
+        """Table of observatory codes from the IAU Minor Planet Center[1].
+
+
+        Parameters
+        ----------
+        get_raw_response : bool, optional
+            Return raw data without parsing into a table (default:
+            `False`).
+
+        cache : bool, optional
+            Cache results or use cached results (default: `True`).
+
+
+        Returns
+        -------
+        tab : Table
+            Table of codes, coordinates, and names.
```

Comment:
This is actually incorrect: the `async` method is returning a `requests` response object, not a Table.  The docstring failure is happening because the `async_to_sync_docstr` processor is failing here... https://github.com/astropy/astroquery/blob/master/astroquery/utils/process_asyncs.py#L56

Reply:
Actually, I'm not certain the failure is related to this, but I suspect so still...

Reply:
textwrap.dedent(doc) did not work as expected on this docstring because I've always started my docstrings just after the triple quote, i.e., not indented!  Because of this the Returns section was not stripped and the new one appended, resulting in two Returns sections and the error.

I'll fix the async section and my docstrings then I think this will pass.